tests/testthat/test_resample_to_field.R

```diff
@@ -0,0 +1,35 @@
+context("resample to field observations")
+
+test_that("run glm simulation", {
+  sim_folder <<- run_example_sim(verbose = F)
+  nc_file <<- file.path(sim_folder, 'output/output.nc')
+  field_file <<- file.path(sim_folder, 'LakeMendota_field_data_hours.csv')
+  field_obs <<- read_field_obs(field_file, var_name = 'temp')
+  expect_is(field_obs, 'data.frame')
+})
+
+test_that('test for non-overlapping simulation and observation dates', {
```

Comment:
I’m not sure what they would be. There currently aren’t any tests of that sort.

Reply:
Understood. The testing in this package isn't great, which is why it makes it a bit challenging to put new methods in because we don't know what all might be changing. This was (I think) the first formal R package I made, so it is pretty clunky when it comes to isolating elements of functions into testable things, hence this issue. But (see my comment below on inner vs left join) we'd want to make sure we know what is coming out of the old vs new implementations and whether it is the same not just in some example data, but in the edge cases too (e.g., from below, when obs are deeper than the lake's max depth).